### PR TITLE
Refactored d2enum

### PR DESCRIPTION
**Pull request by Intyre**

### CODE REVIEW

File: d2common/d2enum/monster_animation_mode.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	MonsterAnimationModeAttack1                               // A1
+	MonsterAnimationModeAttack2                               // A2
+	MonsterAnimationModeBlock                                 // BL
+	MonsterAnimationModeCast                                  // SC
+	MonsterAnimationModeSkill1                                // S1
+	MonsterAnimationModeSkill2                                // S2
+	MonsterAnimationModeSkill3                                // S3
+	MonsterAnimationModeSkill4                                // S4
+	MonsterAnimationModeDead                                  // DD
+	MonsterAnimationModeKnockback                             //
```
Comment (gravestench) [inline d2common/d2enum/monster_animation_mode.go:24]: curious, is `MonsterAnimationModeKnockback` an empty string?

File: d2common/d2enum/player_animation_mode.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	PlayerAnimationModeBlock                                  // BL
+	PlayerAnimationModeCast                                   // SC
+	PlayerAnimationModeThrow                                  // TH
+	PlayerAnimationModeKick                                   // KK
+	PlayerAnimationModeSkill1                                 // S1
+	PlayerAnimationModeSkill2                                 // S2
+	PlayerAnimationModeSkill3                                 // S3
+	PlayerAnimationModeSkill4                                 // S4
+	PlayerAnimationModeDead                                   // DD
+	PlayerAnimationModeSequence                               // GH
```
Comment (gravestench) [inline d2common/d2enum/player_animation_mode.go:28]: are `Sequence` and `Knockback` both GH ?

File: d2common/d2enum/hero_string.go [hunk 3]
```diff
@@ -22,9 +22,9 @@ const _Hero_name = "BarbarianNecromancerPaladinAssassinSorceressAmazonDruid"
 
 var _Hero_index = [...]uint8{0, 0, 9, 20, 27, 35, 44, 50, 55}
 
-func (h Hero) String() string {
-	if h < 0 || h >= Hero(len(_Hero_index)-1) {
-		return "Hero(" + strconv.FormatInt(int64(h), 10) + ")"
+func (i Hero) String() string {
```
Comment (gravestench) [inline d2common/d2enum/hero_string.go:25]: Why was this necessary? Not really a big deal, just curious.

File: d2core/d2object/init_function.go [hunk 4]
```diff
@@ -27,15 +27,15 @@ func initObject(ob *Object) bool {
 
 // Initializes torch/brazier type objects
 func initTorch(ob *Object) {
-	if ob.objectRecord.HasAnimationMode[d2enum.AnimationModeObjectOperating] {
-		ob.setMode("ON", 0, true)
+	if ob.objectRecord.HasAnimationMode[d2enum.ObjectAnimationModeOperating] {
+		ob.setMode(d2enum.ObjectAnimationModeOpened, 0, true)
```
Comment (Ziemas) [inline d2core/d2object/init_function.go:31]: Oops, I guess I messed up those checks when I implemented those.

They should check for and set ObjectAnimationModeOpened

File: d2core/d2object/init_function.go [hunk 5]
```diff
@@ -27,15 +27,15 @@ func initObject(ob *Object) bool {
...
 	}
 }
 
 func initWaypoint(ob *Object) {
 	// Turn these on unconditionally for now, they look nice :)
-	if ob.objectRecord.HasAnimationMode[d2enum.AnimationModeObjectOperating] {
-		ob.setMode("ON", 0, true)
+	if ob.objectRecord.HasAnimationMode[d2enum.ObjectAnimationModeOperating] {
+		ob.setMode(d2enum.ObjectAnimationModeOpened, 0, true)
```
Comment (Ziemas) [inline d2core/d2object/init_function.go:38]: Also here.

**Comment by gravestench** [comment]
@Intyre you may want to update the title of this pull request as you're not only refactoring animation mode enum
